src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ComposeStructuralVariantHaplotypesSpark.java

```diff
@@ -0,0 +1,828 @@
+            alignedOutputWriter.addAlignment(alternativeRecord);
+        }
+        final SAMRecord unmappedReferenceRecord  = unmapAndPointToVariant(referenceRecord.deepCopy(), svc);
+        final SAMRecord unmappedAlternativeRecord = unmapAndPointToVariant(alternativeRecord.deepCopy(), svc);
+        outputWriter.addAlignment(unmappedReferenceRecord);
+        outputWriter.addAlignment(unmappedAlternativeRecord);
+    }
+
+    private String supplementaryAlignmentLikeString(final SAMRecord record) {
+        if (record.getReadUnmappedFlag()) {
+            return new Cigar().toString();
+        } else {
+            return Utils.join(",", record.getReferenceName(), record.getStart(), record.getReadNegativeStrandFlag() ? "-" : "+",
+                                             record.getCigar().toString(), record.getMappingQuality()) + ";";
+        }
+    }
+
+    private SAMRecord unmapAndPointToVariant(final SAMRecord input, final StructuralVariantContext svc) {
+        final SAMRecord record = input.deepCopy();
+        if (record.getReadNegativeStrandFlag()) {
+            record.setReadNegativeStrandFlag(false);
+            final byte[] bases = record.getReadBases();
+            if (bases != null) {
+                SequenceUtil.reverseComplement(bases);
+                record.setReadBases(bases);
+            }
+            final byte[] quals = record.getBaseQualities();
+            if (quals != null) {
+                SequenceUtil.reverseQualities(quals);
+                record.setBaseQualities(quals);
+            }
+        }
+        record.setReadUnmappedFlag(true);
+        record.setReferenceName(svc.getContig());
+        record.setAlignmentStart(svc.getStart());
+        record.setCigar(new Cigar());
+        return record;
+    }
+
+    private <V extends VariantContext> Tuple2<V, List<AlignedContig>> resolvePendingContigs(final Tuple2<V, List<AlignedContig>> vc, final ReadsSparkSource s) {
+        logger.debug("VC " + vc._1().getContig() + ":" + vc._1().getStart() + "-" + vc._1().getEnd());
+        final List<AlignedContig> updatedList = vc._2().stream()
+                .map(a -> {
+                    if (a.contigSequence[0] != 0 && a.contigSequence[a.contigSequence.length - 1] != 0) {
+                        return a;
+                    } else {
+                        final String targetName = a.contigName;
+                        final List<SimpleInterval> locations = a.alignmentIntervals.stream()
+                                .map(ai -> ai.referenceSpan).collect(Collectors.toList());
+                        final GATKRead candidate = s.getParallelReads(alignedContigsFileName, referenceArguments.getReferenceFileName(),
+                                        new TraversalParameters(locations, false))
+                                .filter(rr -> rr.getName().equals(targetName))
+                                .filter(rr -> !rr.getCigar().containsOperator(CigarOperator.H))
```

Comment:
Yeah, I simply wasn't making any assumptions about whether the primary alignment has the whole sequence; but it seems reasonable to do so and thus it will simply things subsantianly